sky/optimizer.py

```diff
@@ -887,4 +883,10 @@ def _fill_in_launchable_resources(
         launchable[resources] = _filter_out_blocked_launchable_resources(
             launchable[resources], blocked_launchable_resources)
 
+        for r in launchable[resources]:
```

Comment:
Q: why move it to here, rather than after L849? Was thinking checking `resources` in that loop makes more sense, as it represents a validation of the user-requested resources.  Here, it may be possible than `launchable[resources]` has more than 1 "expanded" resources, and throwing an error on these may be unexpected?

Reply:
I meant that here launchable[resources] may have more than 1 element, - can some of them pass the check, while some fail?  In these cases it may make sense to remove the candidates that fail rather than raising an error to the whole program.

Reply:
OK. That makes sense. I've rolled back the change.